hbase-client/src/main/java/org/apache/hadoop/hbase/filter/FilterListWithAND.java

```diff
@@ -206,17 +244,28 @@ public boolean filterRowKey(Cell firstRowCell) throws IOException {
     if (isEmpty()) {
       return super.filterRowKey(firstRowCell);
     }
-    boolean retVal = false;
+    boolean anyRowKeyFiltered = false;
+    boolean anyHintingPassed = false;
     for (int i = 0, n = filters.size(); i < n; i++) {
       Filter filter = filters.get(i);
-      if (filter.filterAllRemaining() || filter.filterRowKey(firstRowCell)) {
+      if (filter.filterAllRemaining()) {
```

Comment:
Do we need to check this here? We do have a filterAllRemaining method below?

Reply:
Technically no, but we do it anyway everywhere, so I didn't want to remove it just here.
There is also an MR job that possibly requires this.
I have opened HBASE-28633 for this.

Reply:
I realize that I referred a wrong ticket in the comment.
Fixing.

Reply:
Done